pkg/sbom/node.go

```diff
@@ -16,6 +16,19 @@ import (
 // updates to the node proto should also be reflected in most of these
 // functions as they operate on the Node's fields
 
+func NewNode() *Node {
+	return &Node{
+		Licenses:           []string{},
+		Attribution:        []string{},
+		Suppliers:          []*Person{},
+		Originators:        []*Person{},
+		ExternalReferences: []*ExternalReference{},
+		FileTypes:          []string{},
+		Identifiers:        map[int32]string{},
+		Hashes:             map[int32]string{},
+	}
+}
+
```

Comment:
if people use NewNode(), Hashes is populated.  But we should assume some people (like me) will just happily 

```
	node1 := &sbom.Node{
                SomeField: SomeValue
        }
```

And we should protect against that before trying to use or set any values that assume it has been previously created with the NewNode() constructor

Reply:
100%